package/utils/lua5.3/patches/030-archindependent-bytecode.patch

```diff
@@ -0,0 +1,156 @@
+--- a/src/ldump.c
++++ b/src/ldump.c
+@@ -7,6 +7,8 @@
+ #define ldump_c
+ #define LUA_CORE
+ 
++#include <endian.h>
++
+ #include "lprefix.h"
+ 
+ 
+@@ -74,7 +76,7 @@ static void DumpString (const TString *s
+   if (s == NULL)
+     DumpByte(0, D);
+   else {
+-    size_t size = tsslen(s) + 1;  /* include trailing '\0' */
++    unsigned int size = tsslen(s) + 1;  /* include trailing '\0' */
+     const char *str = getstr(s);
+     if (size < 0xFF)
+       DumpByte(cast_int(size), D);
+@@ -187,12 +189,15 @@ static void DumpHeader (DumpState *D) {
+   DumpByte(LUAC_FORMAT, D);
+   DumpLiteral(LUAC_DATA, D);
+   DumpByte(sizeof(int), D);
+-  DumpByte(sizeof(size_t), D);
++  DumpByte(sizeof(unsigned int), D);
+   DumpByte(sizeof(Instruction), D);
+   DumpByte(sizeof(lua_Integer), D);
+   DumpByte(sizeof(lua_Number), D);
+-  DumpInteger(LUAC_INT, D);
+-  DumpNumber(LUAC_NUM, D);
++#if __BYTE_ORDER == __LITTLE_ENDIAN
++  DumpByte(0, D);
++#else
++  DumpByte(1, D);
++#endif
+ }
+ 
+ 
+--- a/src/llimits.h
++++ b/src/llimits.h
+@@ -159,11 +159,7 @@ typedef LUAI_UACINT l_uacInt;
+ ** type for virtual-machine instructions;
+ ** must be an unsigned with (at least) 4 bytes (see details in lopcodes.h)
+ */
+-#if LUAI_BITSINT >= 32
+ typedef unsigned int Instruction;
+-#else
+-typedef unsigned long Instruction;
+-#endif
+ 
+ 
+ 
+--- a/src/lundump.c
++++ b/src/lundump.c
+@@ -7,6 +7,8 @@
+ #define lundump_c
+ #define LUA_CORE
+ 
++#include <endian.h>
++
+ #include "lprefix.h"
+ 
+ 
+@@ -33,6 +35,7 @@ typedef struct {
+   lua_State *L;
+   ZIO *Z;
+   const char *name;
++  int swap;
+ } LoadState;
+ 
+ 
+@@ -46,13 +49,52 @@ static l_noret error(LoadState *S, const
+ ** All high-level loads go through LoadVector; you can change it to
+ ** adapt to the endianness of the input
+ */
+-#define LoadVector(S,b,n)	LoadBlock(S,b,(n)*sizeof((b)[0]))
++#define LoadVector(S,b,n)	LoadVectorEndianness(S,b,n,sizeof((b)[0]))
+ 
+ static void LoadBlock (LoadState *S, void *b, size_t size) {
+   if (luaZ_read(S->Z, b, size) != 0)
+     error(S, "truncated");
+ }
+ 
++static void LoadVectorEndianness(LoadState *S, void *b, size_t n, size_t size)
```

Comment:
I would try to keep this similar to the referenced patch.
Lets drop the define and define a LoadVector function directly.

Reply:
Done.